braindecode/datasets/datasets.py

```diff
@@ -0,0 +1,446 @@
+                for run_id, raw in sess_data.items():
+                    raws.append(raw)
+                    subject_ids.append(subj_id)
+                    session_ids.append(sess_id)
+                    run_ids.append(run_id)
+        info = pd.DataFrame(zip(subject_ids, session_ids, run_ids),
+                            columns=["subject", "session", "run"])
+        return raws, info
+
+
+def fetch_data_with_moabb(dataset_name, subject_ids):
+    # ToDo: update path to where moabb downloads / looks for the data
+    """
+    Fetch data using moabb.
+
+    Parameters
+    ----------
+    dataset_name: str
+        the name of a dataset included in moabb
+    subject_ids: list(int) | int
+        (list of) int of subject(s) to be fetched
+
+    Returns
+    -------
+    raws: mne.Raw
+    info: pandas.DataFrame
+
+    """
+    dataset = _find_dataset_in_moabb(dataset_name)
+    subject_id = [subject_ids] if isinstance(subject_ids, int) else subject_ids
+    return _fetch_and_unpack_moabb_data(dataset, subject_id)
+
+
+# TODO: EEG / EEGDataset which has a ConcatDataset and additional info
+
+
+class MOABBDataset(ConcatDataset):
+    """
+    A class for moabb datasets.
+
+    Parameters
+    ----------
+    dataset_name: name of dataset included in moabb to be fetched
+    subject_ids: list(int) | int
+        (list of) int of subject(s) to be fetched
+    trial_start_offset_samples: int
+        start offset from original trial onsets in samples
+    trial_stop_offset_samples: int
+        stop offset from original trial onsets in samples
+    supercrop_size_samples: int
+        supercrop size
+    supercrop_stride_samples: int
+        stride between supercrops
+    drop_samples: bool
+        whether or not have a last overlapping supercrop/window, when
+        supercrops/windows do not equally devide the continuous signal
+    ignore_events: bool
+        when True, ignores events specified in mne.Raw and uses a
+        FixedLenthWindower to create supercrops/windows
+
+    """
+    # TODO: include preprocessing at different stages
+    def __init__(
+            self, dataset_name, subject_ids, trial_start_offset_samples,
+            trial_stop_offset_samples, supercrop_size_samples,
+            supercrop_stride_samples, drop_samples=False, ignore_events=False):
+        if ignore_events:
+            windower = FixedLengthWindower
+        else:
+            windower = EventWindower
+        windower = windower(
+            trial_start_offset_samples=trial_start_offset_samples,
+            trial_stop_offset_samples=trial_stop_offset_samples,
+            supercrop_size_samples=supercrop_size_samples,
+            supercrop_stride_samples=supercrop_stride_samples,
+            drop_samples=drop_samples)
+
+        raws, info = fetch_data_with_moabb(dataset_name, subject_ids)
+        all_windows_ds = []
+        for raw_i, raw in enumerate(raws):
+            base_ds = BaseDataset(raw, info.iloc[raw_i])
```

Comment:
I left a comment in the windower script - Why do we first create a BaseDataset out of the Raw object? Wouldn't it be simpler to just pass the Raw object directly to the Windower object?